src/include/firebird/impl/inf_pub.h

```diff
@@ -171,6 +174,16 @@ enum db_info_crypt			/* flags set in fb_info_crypt_state */
 	fb_info_crypt_process = 0x02
 };
 
+enum info_provider_features // Flags set in response to fb_info_provider_features
+{
+	fb_feature_multi_statements 	= 0x00001,  // Multiple prepared statements in single attachment
+	fb_feature_multi_transactions	= 0x00002,	// Multiple concurrent transaction in single attachment
+	fb_feature_named_parameters		= 0x00004,	// Query parameters can be named
+	fb_feature_session_reset		= 0x00008,	// ALTER SESSION RESET is supported
+	fb_feature_read_consistency		= 0x00010,	// Read consistency TIL is supported
+	fb_feature_statement_timeout	= 0x00020	// Statement timeout is supported
+};
+
```

Comment:
I'm not sure I like features being described as a bitmap. What if there will be more than 32 features in FB6? I'd rather consider returning them as a variable-length byte vector in the API. Inside the extds/* modules this vector can be parsed into a int32/int64 bit mask, if you prefer to preserve the current code "as is". This becomes an implementation detail that can be changed when needed.

Reply:
I was thinking about array of bytes but failed to invent a handy and error-prone way to declare the constants so I decided to use int64 bitmask for awhile and if number of features grow over 64 another inf item (fb_info_provider_features2) can be added. If you have a good idea - I'm all ears.

Reply:
Byte vector is IMO preferred here cause it does not have endianess issues. It also limits set of features (255) but sooner of all that's enough. What about declaration - what's bad with trivial:
enum fb_features
{
  fb_feature_1 = 1,
  fb_feature_2,
  fb_feature_3
.......
};

Reply:
Seven bits are wasted. I'd prefer to keep data exchange via network dense. Besides, could you show how declaration of ENGINE_FEATURES constant would look like in this case?

Endianess is solved by isc_portable_integer()/ClumpletReader.